### PR TITLE
Implement overviews support for all dataview types

### PR DESCRIPTION
**Pull request by jgoizueta**
See #466

### CODE REVIEW

**Comment by jgoizueta** [comment]
cc @rochoa can you CR this?

It's been tested in staging, the idea is to further test it in production. Basically overviews are now used for aggregation and histogram dataviews. Cases that cannot be handled well with the current overviews, suc h as minimum/maximum formulas fallback to use the base table.

File: lib/cartodb/models/dataview/overviews/histogram.js [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+
+var columnTypeQueryTpl = dot.template(
+    'SELECT pg_typeof({{=it.column}})::oid FROM ({{=it.query}}) _cdb_histogram_column_type limit 1'
+);
+var columnCastTpl = dot.template("date_part('epoch', {{=it.column}})");
+
+var BIN_MIN_NUMBER = 6;
+var BIN_MAX_NUMBER = 48;
+
+var basicsQueryTpl = dot.template([
```
Comment (rochoa) [inline lib/cartodb/models/dataview/overviews/histogram.js]: Can we try to share some of the templates between different dataviews? Some of the query templates are exactly the same and we only change some parts of the CTE.

For instance in this particular case we are just changing a bit the `basics` queries, and the `histogramQueryTpl` query itself. Apart from that everything else is quite similar.

File: lib/cartodb/models/dataview/overviews/list.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
+var BaseOverviewsWidget = require('./base');
```
Comment (rochoa) [inline lib/cartodb/models/dataview/overviews/list.js]: Use `Dataview` for naming.

File: lib/cartodb/models/dataview/overviews/histogram.js (hunk 1 above)
Comment (jgoizueta) [inline lib/cartodb/models/dataview/overviews/histogram.js]: Yes, although that might diverge in the feature... I'll give it a thought

File: lib/cartodb/models/dataview/overviews/list.js (hunk 2 above)
Comment (jgoizueta) [inline lib/cartodb/models/dataview/overviews/list.js]: s/Widget/Dataview

**Pull request merged**